aspnetcore/migration/webapi.md

```diff
@@ -4,47 +4,167 @@ author: ardalis
+1. In the **Configure your new project** dialog, Name the project *ProductsCore*, and select **Create**.
+1. In the **Create a new ASP.NET Core Web Application** dialog, confirm that **.NET Core** and **ASP.NET Core 3.1** are selected. Select the **API** project template, and select **Create**.
+1. Remove the *WeatherForecast.cs* and *Controllers/WeatherForecastController.cs* example files from the new *ProductsCore* project.
+
+The solution now contains two projects. The following sections explain migrating the *ProductsApp* project's contents to the *ProductsCore* project.
+
+## Migrate configuration
+
+ASP.NET Core doesn't use:
+
+* *App_Start* folder or the *Global.asax* file
+* *web.config* file is added at publish time.
+
+The `Startup` class:
+
+* Replaces *Global.asax*.
+* Handles all app startup tasks.
+
+For more information, see <xref:fundamentals/startup>.
+
+## Migrate models and controllers
+
+The following code shows the `ProductsController` update for ASP.NET Core:
+[!code-csharp[](webapi/sample/3.x/ProductsApp/Controllers/ProductsController.cs)]
+
+Update the `ProductsController` for ASP.NET Core:
+
+1. Copy *Controllers/ProductsController.cs* from the original project to the new one.
+1. Copy the *Models* folder from the original project to the new one.
+1. Change the copied files' root namespace to `ProductsCore`.
+1. Update the `using ProductsApp.Models;` statement to `using ProductsCore.Models;`.
+
+The following components don't exist in ASP.NET Core:
+
+* `ApiController` class
+* `System.Web.Http` namespace
+* `IHttpActionResult` interface
+
+Make the following changes:
+
+1. Change `ApiController` to <xref:Microsoft.AspNetCore.Mvc.ControllerBase>. Add `using Microsoft.AspNetCore.Mvc;` to resolve the `ControllerBase` reference.
+1. Delete `using System.Web.Http;`.
+1. Change the `GetProduct` action's return type from `IHttpActionResult` to `ActionResult<Product>`.
+
+Simplify the `GetProduct` action's `return` statement to the following:
+
+```csharp
+return product;
+```
+
+## Configure routing
+
+All ASP.NET Core project templates include attribute routing in the generated code.
 
-The *ProductsApp* project includes one controller. The controller inherits from `ApiController` and contains two actions:
+The following <xref:Microsoft.AspNetCore.Builder.EndpointRoutingApplicationBuilderExtensions.UseRouting%2A> and <xref:Microsoft.AspNetCore.Builder.EndpointRoutingApplicationBuilderExtensions.UseEndpoints%2A> calls:
 
-[!code-csharp[](webapi/sample/ProductsApp/Controllers/ProductsController.cs?highlight=28,33)]
+* Register route matching and endpoint execution in the [middleware](xref:fundamentals/middleware/index) pipeline.
+* Replace the *ProductsApp* project's *App_Start/WebApiConfig.cs* file.
 
-The `Product` model used by `ProductsController` is a simple class:
+[!code-csharp[](webapi/sample/3.x/ProductsCore/Startup.cs?name=snippet_Configure&highlight=10,14)]
```

Comment:
I recommend moving this code snippet and its association explanation to the numbered list below. It should be the first step in the list because it's required to use routing.

Reply:
@scottaddie, good point, thanks, but there really isn't a real step to take with UseRouting and UseEndPoints.  It is just already generated with the project.

Is this what you were envisioning?

## Configure routing

1. All ASP.NET Core project templates include attribute routing in the generated code.

The following <xref:Microsoft.AspNetCore.Builder.EndpointRoutingApplicationBuilderExtensions.UseRouting%2A> and <xref:Microsoft.AspNetCore.Builder.EndpointRoutingApplicationBuilderExtensions.UseEndpoints%2A> calls:

* Register route matching and endpoint execution in the [middleware](xref:fundamentals/middleware/index) pipeline.
* Replace the *ProductsApp* project's *App_Start/WebApiConfig.cs* file.

[!code-csharp[](webapi/sample/3.x/ProductsCore/Startup.cs?name=snippet_Configure&highlight=10,14)]

1. Mark the `ProductsController` class with the following attributes:

    ```csharp
    [Route("api/[controller]")]
    [ApiController]
    ```

Reply:
@wadepickett Disregard my comment. I forgot that we tell the reader to create a new project.